src/thor/route_action.cc

```diff
@@ -381,26 +381,28 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
     if (temp_paths.empty())
       return false;
     for (auto& temp_path : temp_paths) {
-      // back propagate time information
-      if (!destination->date_time().empty() &&
-          options.date_time_type() != valhalla::Options::invariant) {
-        auto origin_dt =
-            DateTime::offset_date(destination->date_time(),
-                                  reader->GetTimezoneFromEdge(temp_path.back().edgeid, tile),
-                                  reader->GetTimezoneFromEdge(temp_path.front().edgeid, tile),
-                                  -temp_path.back().elapsed_cost.secs);
-        origin->set_date_time(origin_dt);
+      auto out_tz = reader->GetTimezoneFromEdge(temp_path.back().edgeid, tile);
+      auto in_tz = reader->GetTimezoneFromEdge(temp_path.front().edgeid, tile);
+
+      // we add the timezone info if destination is the last location
+      // and add waiting_secs again from the final destination's datetime, so we output the departing
+      // time at intermediate locations, not the arrival time
+      if ((destination->correlation().original_index() == (options.locations().size() - 1) &&
```

Comment:
since its only for the last location, any chance we could pull this special case outside the loop? no worries if its not ergonomic

Reply:
I think the issue with that is that we use the `PathInfo`'s `edgeid` to get the time zone information here, is there a way to get that time zone information outside of the loop?

Reply:
you could get to it via the TripLeg in PBF, but honestly, that's more hassle than it's worth IMHO.

Reply:
wdyt @kevinkreiser ?

Reply:
yep no worries fine here